session4/src/com/itschool/homework/Challenge6.java

```diff
@@ -1,26 +1,40 @@
 package com.itschool.homework;
 
+import java.util.Scanner;
+
 public class Challenge6 {
 
     public static void main(String[] args) {
 
-        int d = 5;
-        int e = 4;
-
-        System.out.println(d>e);
-        System.out.println(d<e);
-        System.out.println(d==e);
+//        int d = 5;
+//        int e = 4;
+//
+//        System.out.println(d>e);
+//        System.out.println(d<e);
+//        System.out.println(d==e);
+//
+//        if (d >= 0 && d <= 100) {
+//            System.out.println("d is in the range");
+//        }
+//
+//        if (e >= -10 && e <= 3) {
+//            System.out.println("This is not supposed to be displayed");
+//        }
+//
+//        if (d >= 2 && d <= 6) {
+//            System.out.println("d is in the range again");
+//        }
 
-        if (d >= 0 && d <= 100) {
-            System.out.println("d is in the range");
-        }
+        Scanner input = new Scanner(System.in);
+        System.out.println("The first number is");
+        int d = input.nextInt();
+        System.out.println("The second number is");
+        int e = input.nextInt();
 
         if (e >= -10 && e <= 3) {
-            System.out.println("This is not supposed to be displayed");
+            System.out.println("e is not in range");
```

Comment:
if `e` is 1 it will say that is not in range which is not true. You need to negate the entire `if` statement with negation operator **!** (exclamation mark) in order to correct this. So, at line 34, you need to have:
`if (!(e >= -10 && e <= 3)) {`